ttnn/cpp/ttnn/tensor/layout/size.hpp

```diff
@@ -0,0 +1,38 @@
+// SPDX-FileCopyrightText: © 2024 Tenstorrent AI ULC
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#pragma once
+
+#include <array>
+
+namespace tt::tt_metal {
+
+class Size final {
```

Comment:
I see this used in TensorLayout -- anywhere else? Consider renaming this if it's supposed to be fixed for 2D.  `Shape2D`?

Reply:
Let me keep it like this for now. I noted the ask, will see how to best make it.